### PR TITLE
[BEAM-13873] [Playground] Increase test coverage for the environment package

### PR DESCRIPTION
**Pull request by pavel-avilov**
[[BEAM-13873]](https://issues.apache.org/jira/browse/BEAM-13873)
Add unit tests to increase the test coverage for the environment package.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by pavel-avilov** [comment]
@KhaninArtur @ilya-kozyrev

File: playground/backend/internal/environment/application_test.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func TestCacheEnvs_Address(t *testing.T) {
 		want   string
 	}{
 		{
-			name: "all success",
+			name: "All success",
```
Comment (daria-malkova) [inline playground/backend/internal/environment/application_test.go:101]: ditto here and below

File: playground/backend/internal/environment/application_test.go [hunk 2]
```diff
@@ -382,3 +382,39 @@ func TestApplicationEnvs_PipelinesFolder(t *testing.T) {
...
+		port     int
+		protocol string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+	}{
+		{
+			name: "All success",
```
Comment (daria-malkova) [inline playground/backend/internal/environment/application_test.go:399]: ditto

File: playground/backend/internal/environment/environment_service_test.go [hunk 3]
```diff
@@ -308,3 +344,109 @@ func Test_getConfigFromJson(t *testing.T) {
...
+	}
+	tests := []struct {
+		name      string
+		args      args
+		want      *BeamEnvs
+		wantErr   bool
+		envsToSet map[string]string
+	}{
+		{
+			name:      "PREPARED_MOD_DIR not be specified in the environment for GO sdk",
```
Comment (daria-malkova) [inline playground/backend/internal/environment/environment_service_test.go:386]: not be - is not

**Review by daria-malkova — CHANGES_REQUESTED** [review]

File: playground/backend/internal/environment/environment_service_test.go (hunk 3 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/environment_service_test.go]: Done

File: playground/backend/internal/environment/application_test.go (hunk 2 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/application_test.go]: Done

File: playground/backend/internal/environment/application_test.go (hunk 1 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/application_test.go]: Done

File: playground/backend/internal/environment/application_test.go [hunk 4]
```diff
@@ -63,7 +63,7 @@ func TestCacheEnvs_CacheType(t *testing.T) {
 		want   string
 	}{
 		{
-			name: "all success",
+			name: "All success",
```
[the remark this answers is not in the document]
Comment (pavel-avilov) [reply playground/backend/internal/environment/application_test.go]: Done

[the remark this answers is not in the document]
Comment (pavel-avilov) [reply playground/backend/internal/environment/application_test.go]: Done

File: playground/backend/internal/environment/environment_service_test.go [hunk 5]
```diff
@@ -46,8 +50,14 @@ func setup() error {
...
-	filePath := filepath.Join(configFolderName, defaultSdk.String()+jsonExt)
-	err = os.WriteFile(filePath, []byte(javaConfig), 0600)
+	javaConfigPath := filepath.Join(configFolderName, defaultSdk.String()+jsonExt)
+	err = os.WriteFile(javaConfigPath, []byte(javaConfig), 0600)
+	goConfigPath := filepath.Join(configFolderName, pb.Sdk_SDK_GO.String()+jsonExt)
+	err = os.WriteFile(goConfigPath, []byte(goConfig), 0600)
+	pythonConfigPath := filepath.Join(configFolderName, pb.Sdk_SDK_PYTHON.String()+jsonExt)
+	err = os.WriteFile(pythonConfigPath, []byte(pythonConfig), 0600)
+	scioConfigPath := filepath.Join(configFolderName, pb.Sdk_SDK_SCIO.String()+jsonExt)
+	err = os.WriteFile(scioConfigPath, []byte(scioConfig), 0600)
```
Comment (AydarZaynutdinov) [inline playground/backend/internal/environment/environment_service_test.go:60]: Move `0600` as a constant.

File: playground/backend/internal/environment/environment_service_test.go [hunk 6]
```diff
@@ -308,3 +344,109 @@ func Test_getConfigFromJson(t *testing.T) {
...
+			envsToSet: map[string]string{beamSdkKey: "SDK_PYTHON"},
+		},
+		{
+			name:      "BeamSdkKey set to SCIO sdk",
+			want:      NewBeamEnvs(pb.Sdk_SDK_SCIO, scioExecutorConfig, modDir, defaultNumOfParallelJobs),
+			wantErr:   false,
+			envsToSet: map[string]string{beamSdkKey: "SDK_SCIO"},
+		},
+		{
+			name:      "NumOfParallelJobsKey is set with with a positive number",
```
Comment (AydarZaynutdinov) [inline playground/backend/internal/environment/environment_service_test.go:419]:
```suggestion
			name:      "NumOfParallelJobsKey is set with a positive number",
```

File: playground/backend/internal/environment/environment_service_test.go [hunk 7]
```diff
@@ -308,3 +344,109 @@ func Test_getConfigFromJson(t *testing.T) {
...
+			want:      NewBeamEnvs(pb.Sdk_SDK_PYTHON, pythonExecutorConfig, modDir, 1),
+			wantErr:   false,
+			envsToSet: map[string]string{beamSdkKey: "SDK_PYTHON", numOfParallelJobsKey: "1"},
+		},
+		{
+			name:      "NumOfParallelJobsKey is set with with a negative number",
```
Comment (AydarZaynutdinov) [inline playground/backend/internal/environment/environment_service_test.go:425]:
```suggestion
			name:      "NumOfParallelJobsKey is set with a negative number",
```

File: playground/backend/internal/environment/environment_service_test.go [hunk 8]
```diff
@@ -308,3 +344,109 @@ func Test_getConfigFromJson(t *testing.T) {
...
+			want:      NewBeamEnvs(pb.Sdk_SDK_PYTHON, pythonExecutorConfig, modDir, defaultNumOfParallelJobs),
+			wantErr:   false,
+			envsToSet: map[string]string{beamSdkKey: "SDK_PYTHON", numOfParallelJobsKey: "-1"},
+		},
+		{
+			name:      "NumOfParallelJobsKey is set with with incorrect value",
```
Comment (AydarZaynutdinov) [inline playground/backend/internal/environment/environment_service_test.go:431]:
```suggestion
			name:      "NumOfParallelJobsKey is set with incorrect value",
```

**Review by AydarZaynutdinov — CHANGES_REQUESTED** [review]

File: playground/backend/internal/environment/environment_service_test.go (hunk 5 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/environment_service_test.go:60]: Done

File: playground/backend/internal/environment/environment_service_test.go (hunk 6 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/environment_service_test.go:419]: Done

File: playground/backend/internal/environment/environment_service_test.go (hunk 7 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/environment_service_test.go]: Done

File: playground/backend/internal/environment/environment_service_test.go (hunk 8 above)
Comment (pavel-avilov) [reply playground/backend/internal/environment/environment_service_test.go]: Done

**Review by AydarZaynutdinov — APPROVED** [review]
LGTM

**Review by daria-malkova — APPROVED** [review]
LGTM

**Comment by pavel-avilov** [comment]
R: @pabloem

**Comment by pabloem** [comment]
LGTM and thanks! I appreciate that y'all are improving tests

**Pull request merged**